src/utilities/path.ts

```diff
@@ -12,8 +13,35 @@ export type Path<
 > = {
   path: T,
   params: P,
+  toString: () => string,
 }
 
-export function path<T extends string, P extends PathParams<T>>(_path: T, _params: Identity<P>): Path<T, P> {
-  throw 'not implemented'
+export function path<T extends string, P extends PathParams<T>>(path: T, params: Identity<P>): Path<T, P> {
+  const optionalParamRegex = /:\?([\w]+)(?=\W|$)/g
+  const requiredParamRegex = /:([\w]+)(?=\W|$)/g
+  const value: Record<string, any> = params
```

Comment:
I'm guessing this `value` would actually be returned as the `params` in the return of this function? So that the response of this has the complete params object?